apps/next-evans/components/core/BookTeaserCard.tsx

```diff
@@ -72,7 +76,7 @@ const BookTeaserCard: React.FC<Props> = (props) => {
             <Link href={props.documentUrl}>
               <Front
                 author={authorName}
-                edition={mostModernEdition(props.editionTypes)}
+                edition={mostModernEdition(editionTypes(props.editions))}
```

Comment:
hmmm... i'm only seeing the usages in this PR, maybe there are a bunch of others, but it seems like almost everywhere you call `mostModernEdition` you end up having an array of editions, and needing to extract the types using your `editionTypes` helper. would it be better to maybe just have `mostModernEdition` take an array of editions directly? maybe you could quick scan the rest of the usages and see if that is the overwhelming majority.

or, if there are a few important call sites where you just have an array of types, perhaps you could overload the function signature to accept _either_ and practice your TS skills.